torch/csrc/nested_tensor/python_nested_tensor.h

```diff
@@ -0,0 +1,442 @@
+#pragma once
```

Comment:
can you comment what the purpose of this file is?

Is there a reason that the python and non-python parts aren't split up?  You want that structure, right?

Reply:
I'll add that towards the end unless you need it for this review, because I'm sure it'll change a bunch until this is ready.